rr_common/src/planner/planner_node.cpp

```diff
@@ -226,27 +222,35 @@ int main(int argc, char** argv) {
     total_planning_time = 0;
     total_plans = 0;
 
-    ROS_INFO("Planner initialized");
+    g_steer_model->Reset(0, ros::Time::now().toSec());
+    g_speed_model->Reset(0, ros::Time::now().toSec());
+
+    g_map_cost_interface->SetMapStale();
+
+    ROS_INFO("planner initialized");
 
     ros::Rate rate(30);
-    is_new_msg = false;
     while (ros::ok()) {
+        rate.sleep();
         ros::spinOnce();
 
-        if (is_new_msg) {
-            is_new_msg = false;
+        g_steer_model->Update(steer_message->angle, ros::Time::now().toSec());
+        g_speed_model->Update(speed_message->speed, ros::Time::now().toSec());
```

Comment:
Shouldn't we update these with our actual current speed and steering, not the theoretical?

Reply:
Perhaps that should be done with the steering model, for platforms where we have steering angle feedback. (To my knowledge, we don't have steering angle feedback on any of our bots yet.) 

Also, the speed model is also being used as a filter on the speed that is published by the node (planner_node.cpp line 139). So the value of the speed model is more of the ideal speed than the predicted speed. I'm actually not sure that line 238 is necessary.

Reply:
Don't we know our speed due to localization for evgp? The main reason I say this is because when we are running the path planner and forcing the speed to 0 (like on start light) then the speed will ramp up pretty high right when the light goes green.

Reply:
I was testing stuff and forcing the speed to 0 and the paths started fairly straight, then the lines were acting weird (very very wobbly) as the speed reached the max and the real speed was still 0.

Reply:
It makes sense to me that there would be serious problems when the planner thinks the vehicle is going way faster or slower than it really is. What if we reset the speed model to the real value whenever we get an update from /chassis_state, and we use the model's predictions in between?

Reply:
I was thinking of using the odometry msg from robot localization that mixes all our sensors. 
I believe that chassis_state will only give encoder data. We can just make it an optional parameter thou in the planner

Reply:
Right, that would be better